crud/common/utils.lua

```diff
@@ -148,6 +151,23 @@ function utils.extract_key(tuple, key_parts)
     return key
 end
 
+function utils.extract_jsonpath_keys(tuple, index_parts, key_def)
+    key_def = key_def or keydef_lib.new(index_parts)
+
+    -- We have case, when we get already extracted values in
+    -- tuple variable, and we will get error like this:
+    -- 'Tuple field [5] required by space format is missing'.
+    local ok, extracted_values = pcall(function()
+        return key_def:extract_key(tuple)
+    end)
```

Comment:
I don't understand you. It looks like a crutch. Also you will recreate closure for each call. It's too expensive for such simple logic. Usage of pcall is too expensive as well.